container-host-files/opt/scf/bin/klog.sh

```diff
@@ -5,192 +5,73 @@ set -e
-lookfor <<EOF
-[ERROR] WSREP: SST failed: 1 (Operation not permitted)
+      # Get the CF logs inside the pod if there are any.
+      if [ "$(get_phase)" != 'Succeeded' ] && check_for_log_dir; then
+        kubectl cp --namespace "${NS}" --container "${CONTAINER}" "${POD}":/var/vcap/sys/log/ "${CONTAINER_DIR}/" 2> /dev/null
+      fi
 
-${MSG}
-EOF
 
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] InnoDB: Cannot open table mysql/gtid_slave_pos from the internal data dictionary of InnoDB
-
-${MSG}
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] WSREP: no nodes coming from prim view, prim not possible
-
-Nodes all shut down improperly, MySQL must be manually rebootstrapped. Will fail
-if the IST patch has not been applied.
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] WSREP: Failed to prepare for incremental state transfer
-
-IST addresses are misconfigured, likely a <= 4.0.1 cluster, manual
-patch must be applied and MySQL must be recovered manually.
-
-Not sure exactly when/why this occurs. Perhaps when the primary node
-has gone away (as was the case here). The only restitution seems to be
-to full recover the mysql node (and ensure the IST patch is done):
-
-monit stop all # Check for lingering processes and kill them
-rm -rf /var/vcap/store/mysql/
-/var/vcap/jobs/mysql/bin/pre-start
-monit start all
-EOF
-
-read -d '' MSG <<EOF || true
-These point to an etcd that came up in a weird way and thinks there
-are other nodes when there are none. Restoration was done by:
-monit stop all
-rm -rf /var/vcap/store/etcd
-# Restart the pod using kubectl
-EOF
-
-# etcd pod /var/vcap/sys/log/etcd/etcd_ctl.err.log
-lookfor <<EOF
-Error:  cannot sync with the cluster using endpoints https://etcd-
-
-${MSG}
-EOF
-
-# etcd pod /var/vcap/sys/log/etcd/etcd.stderr.log
-# duplicate of previous error?
-lookfor <<EOF
-etcdserver: publish error: etcdserver: request timed out
-
-${MSG}
-EOF
-
-# mysql pods /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-SST disabled due to danger of data loss. Verify data and bootstrap the cluster
-
-IST is probably disabled so it's falling back to this, this happens on
-4.0.1 clusters, the fix is complicated as it involves manually
-patching.
-EOF
-
-# consul pod /var/vcap/sys/log/consul_agent/consul_agent.stdout.log
-lookfor <<EOF
-Only one node should be in bootstrap mode, not adding Raft peer.
+      # Get the pod logs - previous may not be there if it was successful on the first run.
+      # Unfortunately we can't get anything past the previous one.
+      kubectl logs "${POD}" --namespace "${NS}" --container "${CONTAINER}" > "${CONTAINER_DIR}/kube.log"
+      kubectl logs "${POD}" --namespace "${NS}" --container "${CONTAINER}" --previous > "${CONTAINER_DIR}/kube-previous.log" 2> /dev/null || true
+      kubectl describe pods "${POD}" --namespace "${NS}" > "${CONTAINER_DIR}/describe-pod.txt"
```

Comment:
This is describing the pod (I don't think describing the container is even a thing), but for some reason we're doing this once per container.

Sorry I missed this in the last round of reviews.

Reply:
Sure this seems redundant, I have moved the statement outside of container loop.